templates/systemd-override.erb

```diff
@@ -1,8 +1,6 @@
-<%- if scope.lookupvar('::osfamily') == 'Gentoo' -%>
+<%- if @os["name"] == 'Gentoo' -%>
 .include /usr/lib64/systemd/system/<%= @service_name %>.service
-<%- elsif scope.lookupvar('::operatingsystem') == 'Fedora' -%>
-.include /lib/systemd/system/<%= @service_name %>.service
-<% else -%>
+<%- elsif (@os["family"] == 'RedHat' and @os["release"]["major"] <= '7') or (@os["name"] == 'Fedora' and @os["release"]["major"] <= '31') -%>
```

Comment:
EL7 also supports drop ins and I'm pretty sure all Fedoras too.

Reply:
Looks like you're correct however I'm not sure when EL7 added support for drop-in dirs. EL 7.9 includes a newer version of systemd which does support drop-in dirs according to the systemd.unit man page.  I can the PR to work with this.

Reply:
I've been using drop ins on EL7 for ages, also before EL 7.9. Never knew otherwise.

Reply:
I've updated the manifest to create drop-in files on RedHat 7 and 8 nodes.  Gentoo will also now do the same.